packages/core-test-kit/src/test-stylable-core.ts

```diff
@@ -0,0 +1,66 @@
+import { testInlineExpects } from './inline-expectation';
+import { Stylable, StylableConfig, StylableExports, StylableMeta } from '@stylable/core';
+import { createMemoryFs } from '@file-services/memory';
+import type { IDirectoryContents, IFileSystem } from '@file-services/types';
+import { isAbsolute } from 'path';
+
+export interface TestOptions {
+    entries: string[];
+    stylableConfig: TestStylableConfig;
+}
+
+export type TestStylableConfig = Omit<
+    StylableConfig,
+    'fileSystem' | `projectRoot` | `resolveNamespace`
+> & {
+    filesystem?: IFileSystem;
+    projectRoot?: StylableConfig['projectRoot'];
+    resolveNamespace?: StylableConfig['resolveNamespace'];
+};
+
+export function testStylableCore(
+    input: string | IDirectoryContents,
+    options: Partial<TestOptions> = {}
+) {
+    // infra
+    const fs =
+        options.stylableConfig?.filesystem ||
+        createMemoryFs(typeof input === `string` ? { '/entry.st.css': input } : input);
+    const stylable = Stylable.create({
+        fileSystem: fs,
+        projectRoot: '/',
+        resolveNamespace: (ns) => ns,
+        ...(options.stylableConfig || {}),
+    });
+
+    // collect sheets
+    const allSheets = fs.findFilesSync(`/`, { filterFile: ({ path }) => path.endsWith(`.st.css`) });
+    const entries = options.entries || allSheets;
+
+    // transform entries
+    const sheets: Record<string, { meta: StylableMeta; exports: StylableExports }> = {};
+    for (const path of entries) {
+        if (!isAbsolute(path || '')) {
+            throw new Error(testStylableCore.errors.absoluteEntry(path));
+        }
+        const meta = stylable.process(path);
+        const { exports } = stylable.transform(meta);
+        sheets[path] = { meta, exports };
+    }
+
+    // inline test
+    for (const path of allSheets) {
+        const meta = stylable.process(path);
+        if (!meta.outputAst) {
+            // ToDo: test
+            stylable.transform(meta);
+        }
+        testInlineExpects({ meta });
+    }
```

Comment:
`entries` could equal `allSheets` so you are looping over there twice. 

```suggestion
    for (const path of entries) {
        if (!isAbsolute(path || '')) {
            throw new Error(testStylableCore.errors.absoluteEntry(path));
        }
        const meta = stylable.process(path);
        const { exports } = stylable.transform(meta);
        
        // inline 
        testInlineExpects({ meta });
                
        sheets[path] = { meta, exports };
    }
```

Reply:
`entries` is not equal to `allSheets` in case `options.entries` is provided and the filesystem contain `.st.css` files that are not referenced in `entries`.

The reason I went with separate phases was to run a process that is close to a real scenario - run analyze _(process)_ on the `entries` dependencies and then run build _(process/transform)_ on the `entries` in the specified order. After this process is complete run the inline tests in any order.

Reply:
Cool, I would add a comment that explains that duplication since it is that clear unless you deep dive into the logic.